### PR TITLE
[FEATURE] MetricListMetricRetriever - 0.18.x

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
* Follow up to #9612, which refactored `ColumnDescriptiveMetricsMetricRetriever` code to parent class. 
* This PR includes the addition of `MetricListMetricRetriever` and associated unit + integration tests. 
* Current PR is against 0.18 branch. Once approved, there will be a subsequent PR against `develop` with identical changes. 

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

### CODE REVIEW

File: great_expectations/experimental/metric_repository/metric_list_metric_retriever.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+                table_metrics,
+                numeric_column_metrics,
+                timestamp_column_metrics,
+                non_numeric_column_metrics,
+            )
+        )
+
+        return bundled_list
+
+    def _get_non_numeric_column_metrics(
```
Comment (rreinoldsc) [inline great_expectations/experimental/metric_repository/metric_list_metric_retriever.py:89]: Can CDMMetricsRetriever's `_get_non_numeric_column_metrics` be abstracted to the ABC, and this method can just call that parent method N times?

File: great_expectations/experimental/metric_repository/metric_list_metric_retriever.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+            aborted_metrics=aborted_metrics,
+        )
+        return TableMetric[List[str]](
+            batch_id=batch_id,
+            metric_name=metric_name,
+            value=value,
+            exception=exception,
+        )
+
+    def _get_table_column_types(self, batch_request: BatchRequest) -> Metric:
```
Comment (rreinoldsc) [inline great_expectations/experimental/metric_repository/metric_list_metric_retriever.py]: same as above

File: great_expectations/experimental/metric_repository/metric_list_metric_retriever.py [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+            aborted_metrics=aborted_metrics,
+        )
+        return TableMetric[int](
+            batch_id=batch_id,
+            metric_name=metric_name,
+            value=value,
+            exception=exception,
+        )
+
+    def _get_table_columns(self, batch_request: BatchRequest) -> Metric:
```
Comment (rreinoldsc) [inline great_expectations/experimental/metric_repository/metric_list_metric_retriever.py]: same as above

File: great_expectations/experimental/metric_repository/metric_list_metric_retriever.py [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+        metrics: List[Metric] = []
+        if MetricTypes.TABLE_ROW_COUNT in metric_list:
+            metrics.append(self._get_table_row_count(batch_request=batch_request))
+        if MetricTypes.TABLE_COLUMNS in metric_list:
+            metrics.append(self._get_table_columns(batch_request=batch_request))
+        if MetricTypes.TABLE_COLUMN_TYPES in metric_list:
+            metrics.append(self._get_table_column_types(batch_request=batch_request))
+        return metrics
+
+    def _get_table_row_count(self, batch_request: BatchRequest) -> Metric:
```
Comment (rreinoldsc) [inline great_expectations/experimental/metric_repository/metric_list_metric_retriever.py]: same as above

File: great_expectations/experimental/metric_repository/metric_list_metric_retriever.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/experimental/metric_repository/metric_list_metric_retriever.py:89]: good suggestion. I'll look into this.

File: great_expectations/experimental/metric_repository/metrics.py [hunk 5]
```diff
@@ -24,7 +24,14 @@
     AbstractSetIntStr = AbstractSet[Union[int, str]]
 
 
-class MetricTypes(str, enum.Enum):
+class MetricTypesMeta(enum.EnumMeta):
+    """Metaclass definition for MetricTypes that allows for membership checking."""
+
+    def __contains__(cls, item):
+        return item in cls.__members__.values()
+
```
Comment (Shinnnyshinshin) [inline great_expectations/experimental/metric_repository/metrics.py:32]: Allow us to use `in` function for `MetricTypes`

File: great_expectations/experimental/metric_repository/column_descriptive_metrics_metric_retriever.py [hunk 6]
```diff
@@ -68,96 +65,14 @@ def get_metrics(self, batch_request: BatchRequest) -> Sequence[Metric]:
...
-        )
-        batch_id, computed_metrics, aborted_metrics = self._compute_metrics(
-            batch_request, table_metric_configs
-        )
-
+    def _calculate_table_metrics(self, batch_request: BatchRequest) -> Sequence[Metric]:
         metrics = [
-            self._get_table_row_count(batch_id, computed_metrics, aborted_metrics),
-            self._get_table_columns(batch_id, computed_metrics, aborted_metrics),
-            self._get_table_column_types(batch_id, computed_metrics, aborted_metrics),
```
Comment (Shinnnyshinshin) [inline great_expectations/experimental/metric_repository/column_descriptive_metrics_metric_retriever.py:83]: This was moved to the `metric_retriever.py` parent class

Comment (rreinoldsc) [reply great_expectations/experimental/metric_repository/column_descriptive_metrics_metric_retriever.py:83]: yessssssssssss

File: great_expectations/experimental/metric_repository/metrics.py (hunk 5 above)
Comment (rreinoldsc) [reply great_expectations/experimental/metric_repository/metrics.py:32]: sweet

File: tests/experimental/metric_repository/test_metric_list_metric_retriever.py [hunk 7]
```diff
@@ -0,0 +1,789 @@
...
+        f"{MetricListMetricRetriever.__module__}.{MetricListMetricRetriever.__name__}._get_numeric_column_names",
+        return_value=["col1", "col2"],
+    ), mock.patch(
+        f"{MetricListMetricRetriever.__module__}.{MetricListMetricRetriever.__name__}._get_timestamp_column_names",
+        return_value=[],
+    ):
+        metrics = metric_retriever.get_metrics(
+            batch_request=mock_batch_request, metric_list=cdm_metrics_list
+        )
+    # why is this not sorted?
```
Comment (rreinoldsc) [inline tests/experimental/metric_repository/test_metric_list_metric_retriever.py]: ?

Comment (Shinnnyshinshin) [reply tests/experimental/metric_repository/test_metric_list_metric_retriever.py]: ha oops. This should go away

**Review by rreinoldsc — APPROVED** [review]

**Pull request merged**